flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultValidator.java

```diff
@@ -100,6 +104,21 @@ private static Optional<String> firstPresent(Optional<String>... errOpts) {
         return Optional.empty();
     }
 
+    private Optional<String> validateDeploymentName(String name) {
+        if (name == null) {
+            return Optional.of("FlinkDeployment name must be define.");
+        } else {
+            Matcher matcher = DEPLOYMENT_NAME_PATTERN.matcher(name);
+            if (!matcher.matches()) {
+                return Optional.of(
+                        String.format(
+                                "The FlinkDeployment meta.name: %s is a invalid value, and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?'",
```

Comment:
Minor correction:

```
The FlinkDeployment name: %s is invalid. It must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name', or 'abc-123')
```

Reply:
Corrected!

Reply:
can you actually take the string that I wrote? I think you only removed the end but I improved the whole message.
meta.name -> name
a invalid value -> invalid.

Fix example: 123-abc does not match the regex, it is not valid

Reply:
I change the message to this:
`The FlinkDeployment name: %s is invalid, must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name',  or 'abc-123'), and the length must be no more than 45 characters.`
More detailed message have been added, including what to start and end with and length requirements.
What do u think?

Reply:
Yes please change the message to what you wrote (you did not change it to this :D)

Reply:
Can you please just copy-paste this into the message so we can avoid a few other iterations on this? 
```
The FlinkDeployment name: %s is invalid, must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name', or 'abc-123'), and the length must be no more than 45 characters.
```

Reply:
Changed.